vignettes/teal-slice-classes.Rmd

```diff
@@ -0,0 +1,231 @@
+`x >= selected[1] & x <= selected[2]`. `Inf` and `NA` values are also handled.
+
+The constructor of `FilterState` has `extract_type` argument which impacts returned call. `extract_type`
+can be unspecified, `"matrix"` or `"list"` and its value corresponds to the type of the variable prefix
+in the returned condition call. For example if `FilterState` is initialized with
+`extract_type = "matrix"` then the variable in the condition call looks like
+`<dataname>[, "<varname>"]`.
+
+### `FilterStateExpr`
+
+```{r echo=FALSE, out.width='33%'}
+knitr::include_graphics("./images/filter_panel/class_details_filter_state_expr.jpg")
+```
+
+Similarly to `FilterState`, `FilterStateExpr` controls a single filter card and returns logical expression. 
+In `FilterState` call is generated automatically based on the `selected` and other attributes, but in `FilterStateExpr` call needs to be specified manually and it must to be a valid R expression.
+
+### `teal_slice`
+
+```{r echo=FALSE, out.width='33%'}
+knitr::include_graphics("./images/filter_panel/class_details_teal_slice.jpg")
+```
+
+`teal_slice` is a `reactiveValues` object storing filter informations. `teal_slice` object is passed directly 
+to `FilterState$initialize` and is stored inside of the `FilterState`  to keep the current state of the filter. Technically, all values used used to generate a call are in `teal_slice`. `FilterState` can be described as 
+a wrapper around `teal_slice` which provides additional methods to handle filter state.
+
+## Making reproducible filter call
+
+### Overview
+
+```{r echo=FALSE, out.width='100%'}
+knitr::include_graphics("./images/filter_panel/get_call_overview.jpg")
+```
+
+Above diagram presents the filter panel classes and their responsibilities when composing filter calls.
+From code perspective, code generation is handled by nested execution of `get_call`. `FilteredData$get_call`
+calls `FilteredDataset$get_call` which calls `FilterStates$get_call` which calls `FilterState$get_call`. 
+Executed functions return:
+
+- `FilterState$get_call()` returns a logical expression.
+- `FilterStates$get_call()` returns a single filter call by gathering logical expressions returned 
+from `FilterState` and combining them by `&` operator grouped by argument name.
+- `FilteredDataset$get_call()` returns the list of calls taken from `FilterStates` object(s).
+- `FilteredData$get_call(<dataname>)` returns the list of calls from specified `FilteredDataset`.
+
+### Example
+
+```{r echo=FALSE, out.width='100%'}
+knitr::include_graphics("./images/filter_panel/get_call_example.jpg")
+```
+
+Calling `datasets$get_call(<dataname>)` in teal modules executes a chain of calls in all filter panel
+classes. Consider a scenario in which:
+
+(1) `FilteredData` has three `FilteredDataset`(s) `ADSL` , `ADTTE`, `MAE`
+
+(2) `CDISCFilteredDataset` contains `data.frame` (`ADSL`) which can be filtered only in one way
```

Comment:
Do we still have `CDISCFilteredDataset`? Or should this be `DefaultFilteredDataset`?